autora/cycle/plot_utils.py

```diff
@@ -472,3 +480,102 @@ def plot_results_panel_3d(
+    Args:
+        cycle: AER Cycle object that has been run
+        x_vals: Test dataset independent values
+        y_true: Test dataset dependent values
+
+    Returns:
+        List of scores by cycle
+    """
+    l_scores = [s.score(x_vals, y_true) for s in cycle.data.theories]
+    return l_scores
+
+
+def cycle_specified_score(
+    scorer: Callable, cycle: Cycle, x_vals: np.ndarray, y_true: np.ndarray, **kwargs
+):
+    """
+    Calculates score for each cycle using specified sklearn scoring function.
+    Args:
+        scorer: sklearn scoring function
+        cycle: AER Cycle object that has been run
+        x_vals: Test dataset independent values
+        y_true: Test dataset dependent values
+        **kwargs: Keyword arguments to send to scoring function
+
+    Returns:
+
+    """
+    # Get predictions
+    if "y_pred" in inspect.signature(scorer).parameters.keys():
+        d_y_pred = _theory_predict(cycle, x_vals, predict_proba=False)
+    elif "y_score" in inspect.signature(scorer).parameters.keys():
+        d_y_pred = _theory_predict(cycle, x_vals, predict_proba=True)
+
+    # Score each cycle
+    l_scores = []
+    for i, y_pred in d_y_pred.items():
+        l_scores.append(scorer(y_true, y_pred, **kwargs))
+
+    return l_scores
+
+
+def plot_cycle_score(
+    cycle: Cycle,
+    X: np.ndarray,
+    y_true: np.ndarray,
+    scorer: Optional[Callable] = None,
+    x_label: str = "Cycle",
+    y_label: Optional[str] = None,
+    figsize: Tuple[float, float] = (6.4, 4.8),
+    scorer_kw: dict = {},
+    plot_kw: dict = {},
+) -> plt.Figure:
+    """
+    Plots scoring metrics of cycle's theories given test data.
+    Args:
+        cycle: AER Cycle object that has been run
+        X: Test dataset independent values
+        y_true: Test dataset dependent values
+        scorer: sklearn scoring function (optional)
+        x_label: Label for x-axis
+        y_label: Label for y-axis
+        figsize: Figure size tuple in inches
+        scorer_kw: Dictionary of keywords for scoring function if scorer is supplied.
+        plot_kw: Dictionary of keywords to pass to matplotlib 'plot' function.
+
+    Returns:
+        matplotlib.figure.Figure
+    """
+
+    # Use estimator's default scoring method if specific scorer is not supplied
+    if scorer is None:
+        l_scores = cycle_default_score(cycle, X, y_true)
+    else:
+        l_scores = cycle_specified_score(scorer, cycle, X, y_true, **scorer_kw)
+
+    # Plotting
+    fig, ax = plt.subplots(figsize=figsize)
+    ax.plot(np.arange(len(cycle.data.theories)), l_scores, **plot_kw)
+
+    # Labeling
+    ax.set_xlabel(x_label)
+    if y_label is None:
+        if scorer is not None:
+            y_label = scorer.__name__
+        else:
+            y_label = "Score"
+    ax.set_ylabel(y_label)
+    ax.xaxis.set_major_locator(MaxNLocator(integer=True))
+
+    # Turn off spines
+    ax.spines.right.set_visible(False)
+    ax.spines.top.set_visible(False)
```

Comment:
I do agree with George that these spines should default off. Maybe it could be conformed to match how the plot_results_panel_2d function handles it (with spines as an argument set to false):

`if not spines:
    ax.spines.right.set_visible(False)
    ax.spines.top.set_visible(False)`

Reply:
Yeah that's the basic idea. At runtime, Matplotlib has all of its default plot parameters stored within an object called rcParams. This controls things like fonts, weights, and other things like which spines are showing. It initializes the object upon import. You can then edit the instance default configurations by changing the attributes stored in the rcParams object. For example: 

```python 
mpl.rcParams['axes.spines.top'] = False
mpl.rcParams['axes.spines.right'] = False
```

Would turn off all top and right spines for all future plots in your instance. 

So one solution would have those lines at the top  of the plot_utils file. That way whenever you import the plotters the rcParams will be set to our defaults. The user can always switch them back on if they know how to access rcParams.  

Alternatively if we wanted more control over the the styling we can create a custom style sheet for the project. Essentially we could create a custom style sheet containing all defaults specific to autora. You would just load the default styles with a command `plt.style.use('autora')` at the top of all of our plotting utilities. This could be bit overkill though. Custom styles sheets are often used when creating figures for a publication with specific style requirements. I've used them to standardize figure styles to conform to company branding.

Reply:
I've refactored to remove the spine removal from within the plotting functions; And I added lines at the to top of the file to change the rcParam spines default to False. This occurs during module import and will turn off spines for all plots generated within the instance. 

I'm not sure if this is the best solution because it could lead to some unexpected behavior in some edge cases. Mostly it would be if a user was working in Jupyter notebook, the runtime instance stays open, they import the functions, and now any other plots they create will be spineless. They will just have to revert the rcParams or set a different style to achieve to bring back the spines. I don't think that's totally unreasonable?

Reply:
Yeah I think that's totally fine. I do not think it's too much to ask for them to learn a bit of matplotlib if they want to modify their plot design. And I definitely like putting the formatting elsewhere like John initially suggested, this feels cleaner and more streamlined. In time, we could think about adding more parameters there so that they are more easily accessible rather than having users write them out themselves. But I feel that is definitely out of the scope for the current PR.